internal/consensus/state_proposaler.go

```diff
@@ -209,6 +213,11 @@ func (p *Proposaler) sendMessages(ctx context.Context, msgs ...Message) {
 }
 
 func (p *Proposaler) verifyProposal(proposal *types.Proposal, rs *cstypes.RoundState) error {
+	if proposal.Height != rs.Height || proposal.Round != rs.Round {
+		return fmt.Errorf("proposal for invalid height/round, proposal height %d, round %d, expected height %d, round %d",
+			proposal.Height, proposal.Round, rs.Height, rs.Round)
+	}
```

Comment:
why add this check?

Reply:
To have more clear error message, otherwise it just reported "error getting proposer" but the core issue was height/round mismatch (detected in tests, main logic checks this, but still)

Reply:
okay, hopefully there are no other side effects.